### PR TITLE
Implemented category translations ala select option translations.

### LINKED ISSUE #462: On the fly category importer not useable for translation when language = store view
In the situation where store views are used as a language category importer (as it is now) cannot be used to translate categories. 

@dweeves and others have written that you'll need to create a root and associate it to a store (not store view) through store management. But this is not possible with store views. They use whatever category root of what store they belong to.

```
store, sku, categories
world_english, NY99, [world_english]/men/shoes
world_danish, NY99, [world_danish]/mænd/sko
```
store-view

This gives the error

On the fly category creator/importer v0.2.5 - Cannot find site root with names:
world_danish

I kind of like how custom swatches work:

```
sku, store, color
NY99, world_danish, rød::[red]
```
The admin name is mentioned in `[]` at the end

Is something like this possible with categories as well since having multiple roots in Magento for store views is not possible?

E.g.

```
store, sku, categories
world_english, NY99, /men/shoes
world_danish, NY99, /mænd::1::1::0::[men]/sko::1::1::0::[shoes]
```

So basically it is the another options like active after the name of the category. It will be a translation rather than a creation of a new tree.

### PR DESCRIPTION
**Pull request by mblarsen**
The current recommend and only way of using magmi to create a
category root for each store (not store view) and then through mange
stores assign it there. That means you are forced to create a store for
each language, however the recommended way to make translations of a store is to create a store view for each language.

You cannot set the category root on store view level. So the idea a root
per language doesn't work in this case.

This PR address that by implementing select option style translations
with this syntax:

```
   catname::[defaultname] or you could say
   danish::[english]
```

Everything works as before in terms of hierarchies:

```
    mænd::[men]/sko::[shoes];;kvinder::[women]/sko::[shoes]
```

Basically it just swaps the default name into the first position and
stores the translate name as an option like `is_anchor` and `is_active`
and so on.

You can combine this with the just mentioned options like expected:

```
    mænd::1::1::0::[men]
```

and even with position (something that wouldn't have worked before):

```
    mænd::1::1::0::[men]::5
```

(Note: due to existing impl. position has to be the last value and you
have to include all three or none of the others)

If you switch between store views in the admin you'll see that it is the
same category tree you are working on. So this is the right way to do
translations.

Fixes #462

### CODE REVIEW

**Comment by mblarsen** [comment]
Also tested with explicit roots

**Comment by dweeves** [comment]
Thx for this useful enhancement !
